### ISSUE TITLE
Reads boolean variable incorrectly

### ISSUE THREAD

**Issue by ouzor** [rstats-db/RMySQL#20]
I have a bit(1) variable in a MySQL database table with some 0 and some 1 values. Reading this into R however gives only 0 values. Tried both RMySQL_0.8-0 and RMySQL_0.9-3 (from GitHub), same result. 

MySQL installed with 'brew install mysql --client-only --universal'
R session info: 

    R version 3.1.2 (2014-10-31)
    Platform: x86_64-apple-darwin13.4.0 (64-bit)

    locale:
    [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

    attached base packages:
    [1] stats     graphics  grDevices utils     datasets  methods   base     

    other attached packages:
    [1] dplyr_0.3.0.2 RMySQL_0.8-0  DBI_0.3.1    

    loaded via a namespace (and not attached):
    [1] assertthat_0.1 magrittr_1.0.1 parallel_3.1.2 Rcpp_0.11.3    tools_3.1.2

**Comment by hadley** [comment]
This should be better in the dev version, but:

> For the MYSQL_TYPE_BIT type code, BIT values are returned into a string buffer, which is why the corresponding C type is char[]. The value represents a bit string that requires interpretation on the client side. To return the value as a type that is easier to deal with, you can cause the value to be cast to integer using either of the following types of expressions:

> ```
> SELECT bit_col + 0 FROM t
> SELECT CAST(bit_col AS UNSIGNED) FROM t
> ```

> To retrieve the value, bind an integer variable large enough to hold the value and specify the appropriate corresponding integer type code.

**Comment by hadley** [comment]
Adapting the example from http://stackoverflow.com/questions/28178897/rmysql-incorrectly-querying-column-with-type-bit

```R
con <- mysqlDefault()
dbGetQuery(con, "CREATE TABLE suppliers ( aId varchar(10) not null, aBit bit not null)")

dbGetQuery(con, 'INSERT INTO suppliers (aId, aBit) VALUES ("First", 0)')
dbGetQuery(con, 'INSERT INTO suppliers (aId, aBit) VALUES ("Second", 1)')
dbReadTable(con, "suppliers")
#>       aId aBit
#> 1   First   00
#> 2  Second   01
```

**Comment by ouzor** [comment]
Thanks! The dev version of RMySQL (0.10.1.9000) works.

**Comment by laurengoodwin-wf** [comment]
This issue is still not fixed in within the CRAN version.

**Comment by mauricewendt** [comment]
Still not fixed in 0.10.9

**Comment by vzemlys** [comment]
Yep, not fixed in CRAN version 0.10.9. Got bitten by it.
